client/pages/home.jsx

```diff
@@ -1,9 +1,232 @@
+      marginBottom: 3
+    },
+    cardCategoryContent: {
+      fontStyle: 'italic',
+      fontSize: '1.3rem',
+      color: '#52616B'
+    },
+    gutter: {
+      marginBottom: 20
+    }
+  };
+});
 
 export default function Home(props) {
+  const classes = useStyles();
+  const [serverData, pullServerData] = useState([]);
+
+  const dateToday = new Date();
+  const defaultYear = dateToday.getFullYear();
+  let defaultMonth = `${dateToday.getMonth() + 1}`;
+  if (defaultMonth.length === 1) {
+    defaultMonth = '0' + defaultMonth;
+  }
+  let defaultDay = `${dateToday.getDate()}`;
+  if (defaultDay.length === 1) {
+    defaultDay = '0' + defaultDay;
+  }
+  const defaultEndPicker = `${defaultYear}-${defaultMonth}-${defaultDay}`;
+  const todayInMilSeconds = dateToday.getTime();
+  const lastWeekInSeconds = todayInMilSeconds - (604800 * 1000);
+
+  const lastWeekDate = new Date(lastWeekInSeconds);
+  const defaultStartYear = lastWeekDate.getFullYear();
+  let defaultStartMonth = `${lastWeekDate.getMonth() + 1}`;
+  if (defaultStartMonth.length === 1) {
+    defaultStartMonth = '0' + defaultStartMonth;
+  }
+  let defaultStartDay = `${lastWeekDate.getDate()}`;
+  if (defaultStartDay.length === 1) {
+    defaultStartDay = '0' + defaultStartDay;
+  }
+  const defaultStartPicker = `${defaultStartYear}-${defaultStartMonth}-${defaultStartDay}`;
+
+  const [endDate, setEndDate] = useState(defaultEndPicker);
+  const [startDate, setStartDate] = useState(defaultStartPicker);
+  const params = new URLSearchParams({ startDate, endDate });
+  const queryString = '?' + params.toString();
+
+  useEffect(() => {
+    let isCanceled = false;
+    const serverAddress = '/api/your/fitness' + queryString;
+    fetch(serverAddress)
+      .then(response => response.json())
+      .then(data => {
+        !isCanceled && pullServerData(data);
+      }, [null, startDate, endDate]);
+    return () => { isCanceled = true; };
+  });
+
+  const returningNumbers = {
+    workouts: serverData.length,
+    workoutTime: 0,
+    caloriesBurned: 0,
+    musclesWorked: {
+      Chest: 0,
+      Back: 0,
+      Arms: 0,
+      Legs: 0
+    }
+  };
+
+  for (let i = 0; i < serverData.length; i++) {
+    returningNumbers.workoutTime = serverData[i].length + returningNumbers.workoutTime;
+    returningNumbers.caloriesBurned = serverData[i].caloriesBurned + returningNumbers.caloriesBurned;
+    if (serverData[i].muscles.includes('Chest')) {
+      returningNumbers.musclesWorked.Chest = returningNumbers.musclesWorked.Chest + 1;
+    }
+    if (serverData[i].muscles.includes('Back')) {
+      returningNumbers.musclesWorked.Back = returningNumbers.musclesWorked.Chest + 1;
+    }
+    if (serverData[i].muscles.includes('Arms')) {
+      returningNumbers.musclesWorked.Arms = returningNumbers.musclesWorked.Chest + 1;
+    }
+    if (serverData[i].muscles.includes('Legs')) {
+      returningNumbers.musclesWorked.Legs = returningNumbers.musclesWorked.Chest + 1;
+    }
+  }
```

Comment:
This is _a lot_ of logic to be re-run every time the component renders. Instead, format the data on the server into the correct shape before sending it to the client.